packages/website/docs/templates.md

```diff
@@ -0,0 +1,221 @@
+    return [
+      {
+        // ...
+        templates: {
+          item({ item, createElement, Fragment }) {
+            return createElement(Fragment, {}, item.name);
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+By default, `createElement` and `Fragment` default to [Preact](https://preactjs.com/)'s `Preact.createElement` (or `h`) and `preact.Fragment`. You can customize these and provide the virtual DOM implementation you prefer.
+
+#### Using the `htm` library
+
+If you're not using a transpiler to build your app, you can still use Autocomplete with the [htm](https://github.com/developit/htm) library, which lets you use a JSX-like syntax directly in the browser.
+
+```js
+import { html } from 'htm/preact';
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources() {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return html`<div>${item.name}</div>`
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+### Rendering a header and footer
+
+In addition to rendering items, you can customize what to display before and after the list of items using the `header` and `footer` templates.
+
+```js
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          header() {
+            return 'Suggestions';
+          },
+          item({ item }) {
+            return `Result: ${item.name}`;
+          },
+          footer() {
+            return 'Footer';
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+### Styling items
+
+Since you're fully controlling the rendered HTML, you can style it the way you want or use any class-based CSS library.
+
+For example, if you're using [Bootstrap](https://getbootstrap.com/):
+
+```js
+/** @jsx h */
+import { h } from 'preact';
+import { autocomplete } from '@algolia/autocomplete-js';
+
+import 'https://cdn.jsdelivr.net/npm/bootstrap@latest/dist/css/bootstrap.min.css';
+
+autocomplete({
+  // ...
+  getSources() {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return <li class="list-group-item">{item.name}</li>;
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+Or [Tailwind CSS](https://tailwindcss.com/):
+
+```js
+/** @jsx h */
+import { h } from 'preact';
+import { autocomplete } from '@algolia/autocomplete-js';
+
+import 'https://unpkg.com/tailwindcss@latest/dist/tailwind.min.css';
+
+autocomplete({
+  // ...
+  getSources() {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return (
+              <li class="py-2 px-4 rounded-sm border border-gray-200">
+                {item.name}
+              </li>
+            );
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+:::note
+
+Autocomplete also provides an [optional default theme](#).
+
+:::
+
+## Reference
+
+### `templates`
```

Comment:
Shouldn't we list the properties here?

- header
- item
- footer

Reply:
Yes, I didn't because it was still in review in #381 but I guess it's close to done, and we can always update later on.

Handled in e580078.